src/Microsoft.AspNet.Razor.Runtime/TagHelperContentWrapperTextWriter.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Implements a <see cref="TextWriter"/> that wraps around <see cref="TagHelperContent"/>.
+    /// </summary>
+    public class TagHelperContentWrapperTextWriter : TextWriter
```

Comment:
suggest exposing an `IBufferEntryCollectionCopyable` interface (shared source in Common) that `StringCollectionTextWriter` (still in MVC.Core) could implement.  only needs to expose a `CopyTo(BufferEntryCollection)` (name not final) usable from `TagHelperExecutionContext.GetChildContentAsync()` or the `Func` we invoke in that method.